File: 05_crm_subscriber_management/utils/content_generation.py
import requests
import csv

def trigger_content_generation(prompt_file="prompt_templates/cover_image_prompts.json"):
    url = "http://127.0.0.1:5001/generate-assets"
    payload = {"prompt_file": prompt_file}
    headers = {"Content-Type": "application/json"}
    response = requests.post(url, json=payload, headers=headers)
    return response.json()

def get_latest_asset():
    try:
        with open("assets/asset_manifest.csv", "r", encoding="utf-8") as f:
            rows = list(csv.DictReader(f))
            if rows:
                return rows[-1]  # Most recent asset
    except Exception as e:
        print("Error reading asset manifest:", e)
    return None

File: 03_scheduling_automation/browser_automation_service/ai-backend/utils/contentGeneration.js
const fetch = require('node-fetch');
const fs = require('fs');
const path = require('path');
const csvParse = require('csv-parse/lib/sync');

async function triggerContentGeneration(promptFile = "prompt_templates/cover_image_prompts.json") {
  const response = await fetch('http://127.0.0.1:5001/generate-assets', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ prompt_file: promptFile })
  });
  return await response.json();
}

function getLatestAsset() {
  try {
    const manifestPath = path.join(__dirname, '../../../../../04_content_generation/assets/asset_manifest.csv');
    const csv = fs.readFileSync(manifestPath, 'utf8');
    const records = csvParse(csv, { columns: true });
    return records[records.length - 1];
  } catch (e) {
    console.error("Error reading asset manifest:", e);
    return null;
  }
}

module.exports = { triggerContentGeneration, getLatestAsset };
